src/main/java/com/openlattice/chronicle/services/download/ParticipantDataIterable.kt

```diff
@@ -0,0 +1,173 @@
+            val edmCacheManager: EdmCacheManager,
+            private val graphApi: GraphApi,
+            private val entitySetIdGraph: EntitySetIdGraph,
+            val srcPropertiesToExclude: Set<FullQualifiedName>,
+            val edgePropertiesToExclude: Set<FullQualifiedName>,
+            val srcMetadata: Map<UUID, EntitySetPropertyMetadata>,
+            val edgeMetadata: Map<UUID, EntitySetPropertyMetadata>,
+            val participantEKID: UUID
+    ) : Supplier<NeighborPage?> {
+        private var page: NeighborPage?
+
+        companion object {
+            private val logger = LoggerFactory.getLogger(NeighborPageSupplier::class.java)
+            private const val MAX_PAGE_SIZE = 10000
+        }
+
+        init {
+            page = NeighborPage(ImmutableMap.of(), null)
+        }
+
+        override fun get(): NeighborPage? {
+            page = try {
+                val searchFilter = EntityNeighborsFilter(
+                        ImmutableSet.of(participantEKID),
+                        Optional.of(ImmutableSet.of(entitySetIdGraph.srcEntitySetId)),
+                        Optional.of(ImmutableSet.of(entitySetIdGraph.dstEntitySetId)),
+                        Optional.of(ImmutableSet.of(entitySetIdGraph.edgeEntitySetId))
+                )
+                graphApi.getPageOfNeighbors(entitySetIdGraph.dstEntitySetId,
+                        PagedNeighborRequest(searchFilter, page!!.bookmark, MAX_PAGE_SIZE))
+            } catch (e: Exception) {
+                logger.error("error retrieving neighbor page", e)
+                null
+            }
+            return page
+        }
+    }
+
+    private class ParticipantDataIterator(private val neighborPageSupplier: NeighborPageSupplier) : Iterator<Map<String, Set<Any>>> {
+        private val lock = ReentrantLock()
+        private var hasMorePages = true
+        private var finishedCurrentPage = false
+        private var currentIndex = 0
+        private var currentPage: NeighborPage? = null
+
+        companion object {
+            private val logger = LoggerFactory.getLogger(ParticipantDataIterable::class.java)
+        }
+
+        override fun hasNext(): Boolean {
+            return hasMorePages
+        }
+
+
+        private fun parseDateTimeValues(values: Set<Any?>, zoneId: ZoneId): Set<Any> {
+            val result: MutableSet<Any> = Sets.newHashSet()
+            values.forEach(Consumer { `val`: Any? ->
+                try {
+                    val parsed = OffsetDateTime.parse(`val`.toString())
+                            .toInstant()
+                            .atZone(zoneId)
+                            .toOffsetDateTime()
+                            .toString()
+                    result.add(parsed)
+                } catch (e: Exception) {
+                    result.add("")
+                }
+            })
+            return result
+        }
+
+        override fun next(): Map<String, Set<Any>> {
+            lock.lock()
```

Comment:
I don't understand the purpose of this lock. It isn't used outside of this function and seems to serve a synchronization mechanism for the method. Use ```@Synchronized``` annotation instead.